### PR TITLE
Implement window position saved-state.

### PR DESCRIPTION
**Pull request by jeremypw**
Most elementary native apps restore the window position as well as size.

### CODE REVIEW

**Comment by jeremypw** [comment]
Conflicts resolved.  Facility to restore window state (including tiled) as well as position added.

File: libcore/Enums.vala [hunk 1]
```diff
@@ -19,6 +19,49 @@
...
+                    return "Marlin.WindowState.TILED_END";
+                case MAXIMIZED:
+                    return "Marlin.WindowState.MAXIMIZED";
+                default:
+                    return "Marlin.WindowState.INVALID";
+            }
+        }
+
+        public static Marlin.WindowState from_gdk_window_state (Gdk.WindowState state, bool start = true) {
+            if ((state & (Gdk.WindowState.MAXIMIZED | Gdk.WindowState.FULLSCREEN)) > 0) {
```
Comment (donadigo) [inline libcore/Enums.vala]: You can use `in` syntax here.

File: libcore/Enums.vala [hunk 2]
```diff
@@ -19,6 +19,49 @@
...
+                return Marlin.WindowState.MAXIMIZED;
+            } else if ((state & (Gdk.WindowState.TILED)) > 0) {
```
Comment (donadigo) [inline libcore/Enums.vala]: Same here.

File: src/View/Window.vala [hunk 3]
```diff
@@ -234,10 +250,7 @@ namespace Marlin.View {
 
 
             window_state_event.connect ((event) => {
-                if ((bool) event.changed_mask & Gdk.WindowState.MAXIMIZED) {
-                    Preferences.settings.set_boolean("maximized",
-                                                     (bool) get_window ().get_state () & Gdk.WindowState.MAXIMIZED);
-                } else if ((bool) event.changed_mask & Gdk.WindowState.ICONIFIED) {
+                if ((bool) event.changed_mask & Gdk.WindowState.ICONIFIED) {
```
Comment (donadigo) [inline src/View/Window.vala]: Same here.

**Comment by jeremypw** [comment]
@donadigo : Changes requested have been made.

**Pull request merged**